girder/utility/acl_mixin.py

```diff
@@ -52,16 +52,15 @@ def load(self, id, level=AccessType.ADMIN, user=None, objectId=True,
         """
         doc = Model.load(self, id=id, objectId=objectId, fields=fields, exc=exc)
 
-        if doc is not None:
+        if not force and doc is not None:
             if doc.get(self.resourceParent):
                 loadType = self.resourceColl
                 loadId = doc[self.resourceParent]
             else:
-                loadType = doc['attachedToType']
-                loadId = doc['attachedToId']
-
-        if not force and doc is not None:
-            self.model(loadType).load(loadId, level=level, user=user, exc=exc)
+                loadType = doc.get('attachedToType')
+                loadId = doc.get('attachedToId')
+            if loadType is not None and loadId is not None:
```

Comment:
One problem is that we call `models.upload.uploadFromFile` with a `parentType=None` and `parent=None` because we are *going* to attach the upload via the `attachedToId` values, but it isn't attached *yet*.  Events that respond to `data.process` need access to the file, but, if we throw an exception here, they fail.

For a brief duration, we have a resource that has no parent nor attached id.  Either we have to let this pass (a mixed-in controlled resource with no owner allows permission, as this code allows), or we have additional logic to the `uploadFromFile` code to attach the file as it is uploaded.

There may be other cases where it is appropriate to have an ownerless resource (but I don't know what those would be).